src/core/markdownASTToString.ts

```diff
@@ -82,35 +82,48 @@ function markdownContentASTToString(nodes: SemanticMarkdownAST[], options?: Conv
                     if (node.type === 'text') {
-
+                        if (!isPrevNodeFormatting && !isNextNodePunctuation && !/^\s/.test(node.content) && markdownString.length > 0 && !/\s$/.test(markdownString)) {
+                            markdownString += ' ';
+                        }
                         markdownString += `${indent}${node.content}`;
-                    } else if (node.type === 'bold') {
-                        markdownString += `**${node.content}**`;
-                    } else if (node.type === 'italic') {
-                        markdownString += `*${node.content}*`;
-                    } else if (node.type === 'strikethrough') {
-                        markdownString += `~~${node.content}~~`;
-                    } else if (node.type === 'link') {
-                        // Check if the link contains only text
-                        if (node.content.length === 1 && node.content[0].type === 'text') {
-                            // Use native markdown syntax for text-only links
-                            markdownString += ` [${node.content[0].content}](${node.href})`;
+                    } else {
+                        let content = '';
+                        if (Array.isArray(node.content)) {
+                            content = markdownContentASTToString(node.content, options, indentLevel);
                         } else {
-                            // Use HTML <a> tag for links with rich content
-                            const linkContent = markdownContentASTToString(node.content, options, indentLevel + 1);
-                            markdownString += ` <a href="${node.href}">${linkContent}</a>`;
+                            content = node.content;
                         }
-                    }
+                        content = content.trim();
 
+                        if (!isPrevNodeFormatting && markdownString.length > 0 && !/\s$/.test(markdownString)) {
+                            markdownString += ' ';
+                        }
+
+                        if (node.type === 'bold') {
+                            markdownString += `**${content}**`;
+                        } else if (node.type === 'italic') {
+                            markdownString += `*${content}*`;
+                        } else if (node.type === 'strikethrough') {
+                            markdownString += `~~${content}~~`;
+                        } else if (node.type === 'link') {
+                            if (node.content.length === 1 && node.content[0].type === 'text') {
+                                markdownString += `[${content}](${node.href})`;
+                            } else {
+                                markdownString += `<a href="${node.href}">${content}</a>`;
+                            }
+                        }
+
+                        // Add space after closing tag if next node is not punctuation
+                        if (!isNextNodePunctuation && nextNode) {
```

Comment:
Not sure this is needed.. the next node should add it's own space in the beginning if it's needed

Reply:
Please explain why this is needed..